upload-api/functions/bridge.js

```diff
@@ -0,0 +1,331 @@
+      nb: task[2]
+    },
+    proofs: [delegation]
+  }))
+  /**
+   * @type {import('@ucanto/interface').ConnectionView<import('@web3-storage/upload-api').Service>}
+   */
+  const connection = connect({
+    id: servicePrincipal,
+    codec: CAR.outbound,
+    channel: HTTP.open({
+      url: serviceURL,
+      method: 'POST'
+    })
+  })
+  // this is an annoying hack to make typescript happy - it wants at LEAST one argument and needs
+  // to be assured that that's the case
+  const [firstInvocation, ...restOfInvocations] = invocations
+  // @ts-ignore multiple issues here
+  return connection.execute(firstInvocation, ...restOfInvocations)
+}
+
+
+
+/**
+ * AWS HTTP Gateway handler for POST / with ucan invocation router.
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ *
+ * @param {import('aws-lambda').APIGatewayProxyEventV2} request
+ * @param {import('../bridge/types').BridgeRequestContext} context
+ */
+export async function handleBridgeRequest(request, context) {
+  try {
+    const { authSecretHeader, authorizationHeader, body, contentType } = parseAwsLambdaRequest(request)
+
+    if (!authorizationHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no Authorization header'
+      }
+    }
+
+    if (!authSecretHeader) {
+      return {
+        statusCode: 401,
+        body: 'request has no X-Auth-Secret header'
+      }
+    }
+
+    if (!body) {
+      return {
+        statusCode: 400,
+        body: 'request has no body'
+      }
+    }
+
+    if (!contentType) {
+      return {
+        statusCode: 400,
+        body: 'request has no Content-Type header'
+      }
+    }
+
+    const parseAuthSecretResult = await parseAuthSecretHeader(authSecretHeader)
+
+    if (parseAuthSecretResult.error) {
+      return {
+        statusCode: 401,
+        body: parseAuthSecretResult.error.message
+      }
+    }
+    const issuer = parseAuthSecretResult.ok
+
+    const parseAuthorizationResult = await parseAuthorizationHeader(authorizationHeader)
+    if (parseAuthorizationResult.error) {
+      return {
+        statusCode: 401,
+        body: parseAuthorizationResult.error.message
+      }
+    }
+    const delegation = parseAuthorizationResult.ok
+
+
+    const parseBodyResult = await parseBody(contentType, body)
+    if (parseBodyResult.error) {
+      return {
+        statusCode: 400,
+        body: parseBodyResult.error.message
+      }
+    }
+    const tasks = parseBodyResult.ok.tasks
+    const receipts = await invokeAndExecuteTasks(
+      issuer,
+      context.serviceDID,
+      context.serviceURL,
+      tasks,
+      delegation
+    )
+
+    const bridgeReceipts = await buildBridgeReceipts(receipts)
+    return bridgeReceipts.ok ? {
+      statusCode: 200,
+      headers: {
+        "Content-Type": "application/json"
```

Comment:
https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-json

Reply:
ah so - when I did this, AWS stopped handling the response correctly, but I think I can tweak it to use their base64 encoding stuff and this should work - will give it a shot!

Reply:
ah, stringifying already fixed it! done!